src/java.base/share/classes/sun/net/www/protocol/https/HttpsClient.java

```diff
@@ -565,6 +566,9 @@ public void afterConnect() throws IOException, UnknownHostException {
                     // will do the spoof checks in SSLSocket.
                     SSLParameters paramaters = s.getSSLParameters();
                     paramaters.setEndpointIdentificationAlgorithm("HTTPS");
+                    if (!(s instanceof SSLSocketImpl)) {
```

Comment:
Please have a comment that the host has been set previously for SSLSocketImpl.

Reply:
done